cdap-docs/tools/doc-cdap-default.py

```diff
@@ -0,0 +1,204 @@
+        self.name = name
+        self.value = value
+        self.description = description
+        self.final = final
+    
+    def __str__(self):
+        if self.final != None:
+            return "%s:\n%s (%s)\n%s" % (self.name, self.value, self.final, self.description)
+        else:
+            return "%s:\n%s\n%s" % (self.name, self.value, self.description)
+
+    def _append(self, name, value):
+        if self.__dict__[name]:
+            self.__dict__[name] = "%s\n%s" % (self.__dict__[name], value)
+        else:
+            self.__dict__[name] = value
+    
+    def rst(self):
+        NAME_START  = '   * - '
+        VALUE_START = '     - '
+        BLOCK_START = '| ``'
+        BLOCK_JOIN  = '``\n       | ``'
+        DESC_START  = '     - '
+        name = "``%s``" % self.name
+        if self.value.find('\n') != -1:
+            value = BLOCK_START + BLOCK_JOIN.join(self.value.split()) + '``'
+        else:
+            value = "``%s``" % self.value
+        rst = "%s%s\n%s%s\n%s%s" % (NAME_START, name,
+                                    VALUE_START, value,
+                                    DESC_START, self.description)
+        return rst
+    
+    def set_attribute(self, name, value):
+        v1 = "%s" % value
+        if name == 'value':
+            v = "\n".join(v1.split())
+        elif name == 'description' and v1 == 'None':
+            v = ''
+        elif name == 'final':
+            v = (v1.lower() == 'true')
+        else:
+            v = " ".join(v1.split())
+        self.__dict__[name] = v
+            
+    def get_attribute(self, name):
+        if name == 'rst':
+            return self._rst()
+        else:
+            return self.__dict__[name]
+    
+    def append_value(self, value):
+        self._append("value", value)
+
+    def append_description(self, description):
+        self._append("description", description)
+    
+
+def load_defaults(filesource):
+    func = 'load_defaults'
+    # Set paths
+    source_path = os.path.dirname(os.path.abspath(__file__))
+    
+    if filesource:
+        xml_path = filesource    
+    else:
+        xml_path = os.path.join(source_path, RELATIVE_PATH, CDAP_DEFAULT_XML)
+    if not os.path.isfile(xml_path):
+        raise Exception(func, "'%s' not a valid path" % xml_path)
+    
+    exclusions_path = os.path.join(source_path, CDAP_DEFAULT_EXCLUSIONS)
+    if not os.path.isfile(exclusions_path):
+        raise Exception(func, "'%s' not a valid path" % exclusions_path)
+    exclusions = [line.rstrip('\n') for line in open(exclusions_path)]
+    
+    tree = ET.parse(xml_path)
+    root = tree.getroot()
+    items = []
+    for property in root:
+        item = Item()
+        for attribute in property:
+            item.set_attribute(attribute.tag, attribute.text)
+        if item.name not in exclusions:
+            items.append(item)
+
+    return sorted(items, key=lambda item: item.name)
```

Comment:
should we sort things here? Properties in ``cdap-default.xml`` may not have alphabetical ordering. They could be ordered by importance (unlikely). Also, related properties may be grouped with each other, even though they may not be alphabetically ordered. For such cases, should we just preserve the ordering from ``cdap-default.xml``?